lib/shared/sdk/index.js

```diff
@@ -52,8 +63,16 @@ const ADAPTORS = [
  *   console.log(drives)
  * })
  */
-exports.scan = (options) => {
-  return Bluebird.all(_.map(ADAPTORS, (adaptor) => {
-    return adaptor.scan(_.get(options, [ adaptor.name ], {}))
+sdk.scan = (options) => {
+  return Bluebird.all(_.map(sdk.adapters, (adapter) => {
+    return new Bluebird((resolve, reject) => {
+      adapter.scan(_.get(options, [ adapter.id ], {}), (error, devices) => {
```

Comment:
I'm thinking this (the options config) would eventually eventually move to something like `sdk.init(options)` or so, which would then initialise the adapters with the given options, like on line 38 here. That'll enable continuous scanning on the adapter's behalf and on its own schedule without having to call `.scan()` on each adapter every time, as well as simpler error handling in one place.

Not sure whether to do that in this PR, or do that in a follow up, as the general set up needs some more thinking and spec-ing wrt the Adapters Spec.

Reply:
I think that we should go for the lowest hanging fruit now, in order to make a release, and then we can take the time to discuss more radical changes.

Reply:
I think if you do this here it'll pave way for https://github.com/resin-io/etcher/pull/1797 and https://github.com/resin-io/etcher/pull/1800 since we can simply add the progress to the devices and `emit` them.